[23022] TripKitUIModule refactor phase3

MichaelReyes opened the pull request:
# Pull Request (PR) Checklist

Thank you for your contribution! Please confirm that you've checked all the boxes below before submitting your PR. Use `[x]` to check a box, e.g., `[x]`, and make sure there's no space around the brackets.

## PR Context

- **Type**: Refactor
- **Issue Link**: [Redmine Issue](https://redmine.buzzhives.com/issues/23022)
- **Risk Factor**: Medium

## Changes

_Describe your changes in detail, highlighting the problem it solves or the feature it adds._

- [TripKitUI] convert InterCityTimePickerViewModel.java to kotlin
- [TripKitUI] convert TripSegmentHelper.java.java to kotlin
- [TripKitUI] remove unused class SortOrders.java
- [TripKitUI] convert BindingConversions.java to kotlin
- [TripKitUI] Deprecate MainThreadBus.java
- [TripKitUI] convert ProviderUtils.java to kotlin
- [TripKitUI] update ServiceStopsProvider and TimetableProvider to adjust for ProviderUtils updates to Kotlin
- [TripKitUI] convert ServiceLineOverlayTask.java.java to kotlin
- [TripKitUI] convert SqlUtils.java.java to kotlin
- [TripKitUI] convert ViewUtils.java.java to kotlin
- [TripKitUI] convert CheckableImageButton.java.java to kotlin
- [TripKitUI] convert TripGoStyleKit.java to kotlin
- [TripKitUI] convert TripKitUI.java to kotlin
- [TripKitUI] add RealTimeChoreographerTest.kt
- [TripKitUI] add RealTimeChoreographerViewModelTest.kt
- [TripKitUI] add CyclingSpeedRepositoryImplTest
- [TripKitUI] add GetSortedTripGroupsWithRoutingStatusTest
- [TripKitUI] add PrioritiesRepositoryImplTest
- [TripKitUI] add QueryLocationResolverTest
- [TripKitUI] add RemindersRepositoryImplTest
- [TripKitUI] add SegmentCameraUpdateMapperTest
- [TripKitUI] add UnitsRepositoryImplTest
- [TripKitUI] add WalkingSpeedRepositoryImplTest
- [TripKitUI] add LocationSearchErrorViewModelTest
- [TripKitUI] add LocationSearchViewModelTest
- [TripKitUI] add SegmentCameraUpdateRepositoryTest
- [TripKitUI] add LocationSearchErrorViewModelTest
- [TripKitUI] add LocationSearchViewModelTest
- [TripKitUI] add SegmentCameraUpdateRepositoryTest
- [TripKitUI] update usage of android.util.Pair to kotlin.Pair
- [TripKitUI] add FetchAndLoadServicesTest
- [TripKitUI] add GoogleAndTripGoSuggestionViewModelTest
- [TripKitUI] add ServiceDetailItemViewModelTest
- [TripKitUI] add ServiceDetailViewModelTest
- [TripKitUI] add DeparturesRepositoryImplTest
- [TripKitUI] add FetchAndLoadTimetableTest
- [TripKitUI] add FetchServiceTest
- [TripKitUI] add GetA2BTimeTest
- [TripKitUI] add GetDirectionTextTest
- [TripKitUI] add GetFrequencyTextTest
- [TripKitUI] add GetModeAccessibilityTest
- [TripKitUI] add GetOrdinaryTimeTest
- [TripKitUI] remove GetServiceSubTitleText.kt
- [TripKitUI] add GetRealtimeTextTest
- [TripKitUI] add GetServiceTertiaryTextTest
- [TripKitUI] add GetServiceTitleTextTest
- [TripKitUI] add GetTimeRangeTextTest
- [TripKitUI] convert LoadServiceTaskCursorCols.java to kotlin
- [TripKitUI] add LoadServiceTaskTest
- [TripKitUI] add RealTimeExtensionsTest
- [TripKitUI] add ServiceRepositoryImplTest
- [TripKitUI] add CreditSourcesOfDataViewModelTest
- [TripKitUI] add TKUIRouteViewModelTest
- [TripKitUI] add CursorToServiceConverterTest
- [TripKitUI] add CursorToStopConverterTest
- [TripKitUI] add GenericListViewModelTest
- [TripKitUI] add LocationHistoryMapperTest
- [TripKitUI] add LocationHistoryRepositoryTest
- [TripKitUI] add TKUIDateTimePickerDialogViewModelTest
- [TripKitUI] tag FavoriteTripsRepository as deprecated
- [TripKitUI] add FavoritesRepositoryImplTest
- [TripKitUI] add GenericWebViewViewModelTest
- [TripKitUI] add WaypointRepositoryImplTest
- [TripKitUI] failing unit test fixes
- [TripKitUI] convert FoursquareGeocoder.java to kotlin
- [TripKitUI] add AutoCompleteTaskTest
- [TripKitUI] add FetchFoursquareLocationsImplTest
- [TripKitUI] add GetTripFromWaypointsImplTest
- [TripKitUI] add FetchGoogleLocationsImplTest
- [TripKitUI] add FetchTripGoLocationsImplTest
- [TripKitUI] add GeocoderTest.kt
- [TripKitUI] add GeocoderLiveTest
- [TripKitUI] add GoogleGeocoderLiveTest
- [TripKitUI] add DirectionsTripPreviewItemViewModelTest.kt
- [TripKitUI] add DrtItemViewModelTest
- [TripKitUI] add DrtTicketViewModelTest
- [TripKitUI] add ExternalActionTripPreviewItemViewModelTest
- [TripKitUI] add ExternalActionViewModelTest
- [TripKitUI] add InfoGroupViewModelTest
- [TripKitUI] add InterCityTimePickerViewModelTest
- [TripKitUI] add LocationPointerViewModelTest
- [TripKitUI] add ModeLocationTripPreviewViewModelTest
- [TripKitUI] add NearbyTripPreviewItemViewModelTest
- [TripKitUI] add OccupancyViewModelTest
- [TripKitUI] add ResultAggregatorImplTest
- [TripKitUI] add RoutingTimeViewModelMapperTest
- [TripKitUI] add ServiceAlertViewModelTest
- [TripKitUI] update MockKTest to add scheduler init and teardown for coroutines
- [TripKitUI] update viewModel class implementations of me.tatarka.bindingcollectionadapter2.ItemBinding to convert them to lazy to avoid initialization error on unit tests

## Checklist for Reviewers

### Documentation and Code Quality
- [ ] **KDocs Documentation**: Are all changes, new functionalities, and classes documented with KDocs?
- [ ] **Architectural Patterns**: Is there consistent and proper use of architectural patterns (e.g., MVVM, MVP)?

### Testing and Reliability
- [ ] **Unit Testing**: Are there unit tests for all new functionalities and classes?
- [ ] **Emulator and Real Device Testing**: Has the application been tested on both emulators and real devices to ensure compatibility?

### Error Handling and Logging
- [ ] **Error Handling**: Are errors and exceptions caught and handled gracefully, ensuring the app remains stable?
- [ ] **Logging**: Is there proper logging in place for critical errors and information, aiding in debugging and monitoring?


## Testing Procedure

_If applicable, provide steps or commands for testing your changes. This can help reviewers and testers._

- 
- 
-

## Work-in-Progress (WIP)

_List any remaining work or areas that need additional focus. This section can be updated as the work progresses._

- [ ] 
- [ ] 

_Remember to keep this template updated based on the feedback and evolving project standards._

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
[23022]
- [TripKitUI] convert InterCityTimePickerViewModel.java to kotlin

MichaelReyes committed:
[23022]
- [TripKitUI] convert TripSegmentHelper.java.java to kotlin
- [TripKitUI] remove unused class SortOrders.java

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
[23022]
- [TripKitUI] convert BindingConversions.java to kotlin
- [TripKitUI] Deprecate MainThreadBus.java
- [TripKitUI] convert ProviderUtils.java to kotlin
- [TripKitUI] update ServiceStopsProvider and TimetableProvider to adjust for ProviderUtils updates to Kotlin

MichaelReyes committed:
[23022]
- [TripKitUI] convert ServiceLineOverlayTask.java.java to kotlin

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
[23022]
- [TripKitUI] convert SqlUtils.java.java to kotlin
- [TripKitUI] convert ViewUtils.java.java to kotlin

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
[23022]
- [TripKitUI] convert CheckableImageButton.java.java to kotlin

MichaelReyes committed:
[23022]
- [TripKitUI] convert TripGoStyleKit.java to kotlin
- [TripKitUI] convert TripKitUI.java to kotlin

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
[23022]
- [TripKitUI] add RealTimeChoreographerTest.kt
- [TripKitUI] add RealTimeChoreographerViewModelTest.kt

MichaelReyes committed:
[23022]
- [TripKitUI] add CyclingSpeedRepositoryImplTest
- [TripKitUI] add GetSortedTripGroupsWithRoutingStatusTest
- [TripKitUI] add PrioritiesRepositoryImplTest
- [TripKitUI] add QueryLocationResolverTest
- [TripKitUI] add RemindersRepositoryImplTest
- [TripKitUI] add SegmentCameraUpdateMapperTest
- [TripKitUI] add UnitsRepositoryImplTest
- [TripKitUI] add WalkingSpeedRepositoryImplTest

MichaelReyes committed:
[23022]
- [TripKitUI] add LocationSearchErrorViewModelTest
- [TripKitUI] add LocationSearchViewModelTest
- [TripKitUI] add SegmentCameraUpdateRepositoryTest

MichaelReyes committed:
[23022]
- [TripKitUI] add LocationSearchErrorViewModelTest
- [TripKitUI] add LocationSearchViewModelTest
- [TripKitUI] add SegmentCameraUpdateRepositoryTest

MichaelReyes committed:
[23022]
- [TripKitUI] update usage of android.util.Pair to kotlin.Pair
- [TripKitUI] add FetchAndLoadServicesTest
- [TripKitUI] add GoogleAndTripGoSuggestionViewModelTest
- [TripKitUI] add ServiceDetailItemViewModelTest
- [TripKitUI] add ServiceDetailViewModelTest

MichaelReyes committed:
[23022]
- [TripKitUI] add DeparturesRepositoryImplTest
- [TripKitUI] add FetchAndLoadTimetableTest

MichaelReyes committed:
[23022]
- [TripKitUI] add FetchServiceTest
- [TripKitUI] add GetA2BTimeTest
- [TripKitUI] add GetDirectionTextTest
- [TripKitUI] add GetFrequencyTextTest
- [TripKitUI] add GetModeAccessibilityTest
- [TripKitUI] add GetOrdinaryTimeTest

MichaelReyes committed:
[23022]
- [TripKitUI] remove GetServiceSubTitleText.kt
- [TripKitUI] add GetRealtimeTextTest
- [TripKitUI] add GetServiceTertiaryTextTest
- [TripKitUI] add GetServiceTitleTextTest
- [TripKitUI] add GetTimeRangeTextTest

MichaelReyes committed:
[23022]
- [TripKitUI] convert LoadServiceTaskCursorCols.java to kotlin
- [TripKitUI] add LoadServiceTaskTest

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
[23022]
- [TripKitUI] add RealTimeExtensionsTest
- [TripKitUI] add ServiceRepositoryImplTest

MichaelReyes committed:
[23022]
- [TripKitUI] add CreditSourcesOfDataViewModelTest
- [TripKitUI] add TKUIRouteViewModelTest

MichaelReyes committed:
[23022]
- [TripKitUI] add CursorToServiceConverterTest
- [TripKitUI] add CursorToStopConverterTest
- [TripKitUI] add GenericListViewModelTest
- [TripKitUI] add LocationHistoryMapperTest
- [TripKitUI] add LocationHistoryRepositoryTest
- [TripKitUI] add TKUIDateTimePickerDialogViewModelTest

MichaelReyes committed:
[23022]
- [TripKitUI] tag FavoriteTripsRepository as deprecated
- [TripKitUI] add FavoritesRepositoryImplTest
- [TripKitUI] add GenericWebViewViewModelTest
- [TripKitUI] add WaypointRepositoryImplTest

MichaelReyes committed:
[23022]
- [TripKitUI] failing unit test fixes

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
[23022]
- [TripKitUI] convert FoursquareGeocoder.java to kotlin
- [TripKitUI] add AutoCompleteTaskTest
- [TripKitUI] add FetchFoursquareLocationsImplTest
- [TripKitUI] add GetTripFromWaypointsImplTest

MichaelReyes committed:
[23022]
- [TripKitUI] add FetchGoogleLocationsImplTest
- [TripKitUI] add FetchTripGoLocationsImplTest

MichaelReyes committed:
[23022]
- [TripKitUI] add GeocoderTest.kt
- [TripKitUI] add GeocoderLiveTest
- [TripKitUI] add GoogleGeocoderLiveTest

MichaelReyes committed:
[23022]
- [TripKitUI] add DirectionsTripPreviewItemViewModelTest.kt
- [TripKitUI] add DrtItemViewModelTest
- [TripKitUI] add DrtTicketViewModelTest
- [TripKitUI] add ExternalActionTripPreviewItemViewModelTest
- [TripKitUI] add ExternalActionViewModelTest
- [TripKitUI] add InfoGroupViewModelTest
- [TripKitUI] add InterCityTimePickerViewModelTest
- [TripKitUI] add LocationPointerViewModelTest
- [TripKitUI] add ModeLocationTripPreviewViewModelTest
- [TripKitUI] add NearbyTripPreviewItemViewModelTest
- [TripKitUI] add OccupancyViewModelTest
- [TripKitUI] add ResultAggregatorImplTest
- [TripKitUI] add RoutingTimeViewModelMapperTest
- [TripKitUI] add ServiceAlertViewModelTest
- [TripKitUI] update MockKTest to add scheduler init and teardown for coroutines
- [TripKitUI] update viewModel class implementations of me.tatarka.bindingcollectionadapter2.ItemBinding to convert them to lazy to avoid initialization error on unit tests

MichaelReyes committed:
[23022]
- [TripKitUI] add DirectionsTripPreviewItemViewModelTest.kt
- [TripKitUI] add DrtItemViewModelTest
- [TripKitUI] add DrtTicketViewModelTest
- [TripKitUI] add ExternalActionTripPreviewItemViewModelTest
- [TripKitUI] add ExternalActionViewModelTest
- [TripKitUI] add InfoGroupViewModelTest
- [TripKitUI] add InterCityTimePickerViewModelTest
- [TripKitUI] add LocationPointerViewModelTest
- [TripKitUI] add ModeLocationTripPreviewViewModelTest
- [TripKitUI] add NearbyTripPreviewItemViewModelTest
- [TripKitUI] add OccupancyViewModelTest
- [TripKitUI] add ResultAggregatorImplTest
- [TripKitUI] add RoutingTimeViewModelMapperTest
- [TripKitUI] add ServiceAlertViewModelTest
- [TripKitUI] update MockKTest to add scheduler init and teardown for coroutines
- [TripKitUI] update viewModel class implementations of me.tatarka.bindingcollectionadapter2.ItemBinding to convert them to lazy to avoid initialization error on unit tests

MichaelReyes committed:
[23022]
- [TripKitUI] add ActionButtonViewModelTest.kt
- [TripKitUI] add TripPreviewHeaderViewModelTest
- [TripKitUI] add TripPreviewPagerItemViewModelTest
- [TripKitUI] add TripPreviewPagerViewModelTest
- [TripKitUI] add TripPreviewParentViewModelTest
- [TripKitUI] add TripResultListViewModelTest
- [TripKitUI] add TripResultPagerViewModelTest
- [TripKitUI] add TripSegmentGetOffAlertDetailViewModelTest
- [TripKitUI] add TripSegmentGetOffAlertsViewModelTest
- [TripKitUI] add TripSegmentSummaryItemViewModelTest
- [TripKitUI] update GetAlternativeTripForAlternativeService, TripPreviewPagerViewModel to add function to pass Scheduler, to be used to control rx on testing
- [TripKitUI] update TripPreviewHeaderViewModel, TripResultListViewModel, TripSegmentGetOffAlertsViewModel to set ItemBinding initialization to use lazy
- [TripKitUI] deprecate TripPreviewPagerFragmentViewModel
- [TripKitUI] convert WaypointTask.java to kotlin

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
Rename .java to .kt

MichaelReyes committed:
[23022]
- [TripKitUI] convert BearingMarkerIconBuilder.java to kotlin
- [TripKitUI] convert FragmentScope.java to kotlin
- [TripKitUI] Delete deprecated LocationTag.java
- [TripKitUI] add TripResultTransportItemViewModelTest.kt
- [TripKitUI] add TripResultTripViewModelTest
- [TripKitUI] add TripResultViewModelTest
- [TripKitUI] add TripSegmentViewModelTest

MichaelReyes committed:
[23022]
- [TripKitUI] unit test fixes

MichaelReyes committed:
[23022]
- [TripKitUI] update tripkit-android submodule reference

MichaelReyes committed:
Merge branch 'refs/heads/develop' into task/23022-tripkitui-refactor-phase3

# Conflicts:
#	tripkit-android

MichaelReyes committed:
[23022]
- [TripKitUI] update tripkit-android submodule reference

MichaelReyes committed:
[23022]
- [TripKitUI] update tripkit-android submodule reference
- [TripKitUI] unit test fixes

MichaelReyes committed:
[23022]
- [TripKitUI] unit test fixes

MichaelReyes committed:
Merge branch 'refs/heads/develop' into task/23022-tripkitui-refactor-phase3

# Conflicts:
#	TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/tripresults/TripResultListViewModel.kt
#	tripkit-android

MichaelReyes committed:
[23022]
- [TripKitUI] update GetRealtimeText to change early arrival color
- [TripKitUI] update ServiceViewModel to convert ObservableFields to MutableLiveData
- [TripKitUI] update ServiceViewModel to convert ObservableFields to MutableLiveData
- [TripKitUI] update TimetableFragment to convert ObservableFields to MutableLiveData
- [TripKitUI] update TimetableViewModel to convert ObservableFields to MutableLiveData, also convert itemBinding initialization to use lazy
- [TripKitUI] update TripResultListFragment and TripResultListViewModel to change startLocationListener handling to check liveData value and avoid re-triggering of the listener
- [TripKitUI] update TripSegmentItemViewModel to update validation for delay and convert ObservableFields to MutableLiveData

MichaelReyes committed:
[23022]
- [TripKit] update RealTimeStatus to add CANCELLED status
- [TripKitUI] update GetRealtimeText to handle and show CANCELLED status
- [TripKitUI] update ServiceViewModel to add countDownTimeTextColor to set color if status is cancelled
- [TripKitUI] update ServiceViewModelImpl to handle and set cancelled status for cancelled service
- [TripKitUI] update TimetableEntry to add field to handle isCancelled

MichaelReyes committed:
[23022]
- [TripKitUI] unit tests fixes

MichaelReyes committed:
[23022]
- [TripGov5] update NextUpItem to update accessing data from ObservableField to MutableLiveData
- [TripKitUI] update ActionButtonHandler to change ObservableFields to MutableLiveData
- [TripKitUI] update CustomBinding.kt to add setting of ImageView and TextView color if disabled flag is true
- [TripKitUI] update TimetableFragment to add binding lifecycleOwner for LiveData changes observer
- [TripKitUI] update trip_result_segment_item.xml to set minimum height and add strikethrough view for cancelled segments
- [TripKitUI] update trip_segment.xml to add segment cancelled layout and fix segment line positioning
- [TripKitUI] update trip_segment_list_fragment.xml to add a strikethrough view for cancelled trip handling
- [TripKitUI] update TripPreviewHeaderFragment to update value fetching from ObservableField to LiveData
- [TripKitUI] update TripPreviewHeaderViewModel to update value fetching from ObservableField to LiveData
- [TripKitUI] update TripResultListFragment to add binding lifecycleOwner for LiveData changes observer
- [TripKitUI] update TripResultViewModel to change ObservableFields to MutableLiveData and update it usage. Also add handling value for cancelled trip
- [TripKitUI] update TripSegmentItemViewModel and add handling for cancelled segment
- [TripKitUI] update TripSegmentListFragment to add binding lifecycleOwner for LiveData changes observer
- [TripKitUI] update TripSegmentSummaryItemViewModel to change ObservableFields to MutableLiveData
- [TripKitUI] update TripSegmentsViewModel to change ObservableFields to MutableLiveData and set value for cancelled trip
- [TripKitUI] update TripSegmentViewModel to change ObservableFields to MutableLiveData

MichaelReyes committed:
[23022]
- [TripKitUI] unit test fixes
- [TripGov5] update tripkit-android-ui submodule reference

sg-jsonjuliane commented on TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/TripGoStyleKit.kt:
```diff
@@ -0,0 +1,606 @@
+
+        // Local Variables
+        val fillStart = if (fraction == 1f) 0f else asin((fraction * 2f - 1f).toDouble())
+            .toFloat() * 180f / Math.PI.toFloat()
```
Can be defined as `private const val RAD_TO_DEG = 180f / Math.PI.toFloat()`.

Improves readability and avoids repeated computation.

MichaelReyes replied:
Hi @sg-jsonjuliane, this is now updated. Please check again. Thanks 🙇

sg-jsonjuliane commented on TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/TripKitUI.kt:
```diff
@@ -0,0 +1,344 @@
+        PrioritiesRepositoryModule::class,
+        GetRoutingConfigModule::class,
+        BookingModule::class,
+        SchedulerFactoryModule::class,
```
Duplicate entry.

MichaelReyes replied:
Hi @sg-jsonjuliane, this is now updated. Please check again. Thanks 🙇

sg-jsonjuliane commented on TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/geocoding/FoursquareGeocoder.kt:
```diff
@@ -0,0 +1,163 @@
+        }
+
+    companion object {
+        private const val FOURSQUARE_CLIENT_ID = "0QZSSYNBJL1SC3KG45OIO41PIMQIHEB10V2HBSBMUGLZMVYZ"
```
Hardcoded API keys.

MichaelReyes replied:
Hi @sg-jsonjuliane, this is now updated. Please check again. Thanks 🙇

sg-jsonjuliane commented on TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/trippreview/TripPreviewPagerFragmentViewModel.kt:
```diff
@@ -19,7 +19,7 @@ import timber.log.Timber
 import javax.inject.Provider
 
-
+@Deprecated("")
```
It would be ideal to add details, but not necessary.

sg-jsonjuliane commented on TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/tripresult/WaypointTask.kt:
```diff
@@ -0,0 +1,265 @@
+                        jsonSegment.addProperty(KEY_START, waypoint.coordinateString)
+
+                        // We only change once.
+                        changeNextDeparture = true
```
Isn't this redundant? Since the logic checks if the value is true already.

sg-jsonjuliane commented on TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/utils/CustomBinding.kt:
```diff
@@ -573,3 +583,27 @@ fun setLayoutAppBackgroundTint(view: View, @ColorInt default: Int?) {
+                view.setTextColor(disabledColor)
+            } else if (originalColor != null) {
+                view.setTextColor(originalColor)
+            }
```
No else statement / fallback intended?

MichaelReyes replied:
Hi @sg-jsonjuliane, yes, since we won't have anything to set if `originalColor` is null. Thanks.

MichaelReyes committed:
[23022]
- [TripGov5] update application classes and use new config for foursquare client id and secret setting
- [TripKitUI] update FoursquareGeocoder to fetch id and secret key from configs
- [TripKit] update Key.kt to add FourSquareKey data class

MichaelReyes committed:
[23022]
- [TripGov5] update tripkit-android-ui submodule reference
- [TripKitUI] update tripkit-android submodule reference

MichaelReyes committed:
[23022]
- [TripKitUI] update TimetableViewModel to disable auto scrolling

MichaelReyes committed:
[23022]
- [TripKitUI] update tripkit-android submodule reference
